src/Api/Response/Response.php

```diff
@@ -172,6 +161,52 @@ public function getQueryInfoMessage(): string
         }
     }
 
+    public function getSmartDidYouMean(): string
+    {
+        $dataQueryInfoMessage = $this->getData(self::DATA_QUERY_INFO_MESSAGE);
+
+        if (empty($dataQueryInfoMessage)) {
+            return '';
+        }
+
+        $type = !empty($dataQueryInfoMessage['didYouMeanQuery'])
+            ? 'did-you-mean' : $dataQueryInfoMessage['queryStringType'];
+        $alternativeQuery = !empty($dataQueryInfoMessage['didYouMeanQuery'])
+            ? $dataQueryInfoMessage['didYouMeanQuery']
+            : $dataQueryInfoMessage['currentQuery'];
+        $originalQuery = !empty($dataQueryInfoMessage['didYouMeanQuery'])
+            ? $dataQueryInfoMessage['currentQuery']
+            : $dataQueryInfoMessage['originalQuery'];
```

Comment:
This is hard to read and three checks for `!empty($dataQueryInfoMessage['didYouMeanQuery'])`.
Just check it once and set all default values when empty. Maybe a comment would help or renaming variables since the logic is not clear when just reading it.

Reply:
Good improvement, nice idea. Thanks!